### PR TITLE
B(1) = +½ set 8: Espinosa–Moll polygamma

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
#### References to other Issues or PRs

Part 8 of the changes formerly included in #23926 (and the last part committed there before I started splitting it into several PRs), building upon #23985.

#### Brief description of what is fixed or changed

This PR generalises `polygamma()` to the generalised polygamma function described in Espinosa and Moll:
$$\psi(s,z)=\frac{\zeta'(s+1,z)+(\psi(-s)+\gamma)\zeta(s+1,z)}{\Gamma(-s)}$$

#### Release Notes

* functions
    * `polygamma()` now accepts complex arguments, implementing the generalised polygamma function described in [Espinosa and Moll (2004)](http://www.math.tulane.edu/~vhm/papers_html/genoff.pdf).
    * BREAKING: As a result of using the Espinosa–Moll generalisation, `polygamma(-1, x)` now differs from `loggamma(x)` by a constant instead of being the same.

### CODE REVIEW

File: sympy/functions/special/gamma_functions.py [hunk 1]
```diff
@@ -657,95 +663,23 @@ class polygamma(Function):
...
                 if z != nz:
                     return polygamma(n, nz)
-
             if n.is_positive:
                 if z is S.Half:
                     return S.NegativeOne**(n + 1)*factorial(n)*(2**(n + 1) - 1)*zeta(n + 1)
-
             if n is S.NegativeOne:
-                return loggamma(z)
+                return loggamma(z) - log(2*pi) / 2
```
Comment (oscarbenjamin) [inline sympy/functions/special/gamma_functions.py:682]: Is this a change in convention or does it just fix a wrong result?

Comment (oscarbenjamin) [reply sympy/functions/special/gamma_functions.py:682]: I tracked the current code down to #543 but that's such a massive PR that it's hard to understand exactly what's going on.

**Comment by oscarbenjamin** [comment]
I get an error from mpmath for `polygamma(1.5, -1.0)`. I guess that should just be `zoo`.

File: sympy/functions/special/gamma_functions.py (hunk 1 above)
Comment (Parcly-Taxel) [reply sympy/functions/special/gamma_functions.py:682]: It is indeed a change in convention. As noted in the first commit message, when Mathematica runs `PolyGamma[-1, x]` it uses a constant of integration of $0$ in $\psi(-1,x)=\int\psi(0,x)+K$, but that does not lead to a "balanced" function as defined by Espinosa & Moll.

**Comment by oscarbenjamin** [comment]
I think this looks good. Does anyone else have any opinions about this definition of `polygamma`?

**Comment by oscarbenjamin** [comment]
Okay, let's get it in!

**Pull request merged**